google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -278,6 +303,88 @@ static String getRoutingHeader(String streamName, String location) {
     return project + "locations/" + location;
   }
 
+  private String getTableName() {
+    Matcher tableMatcher = streamPatternTable.matcher(this.streamName);
+    return tableMatcher.find() ? tableMatcher.group(1) : "";
+  }
+
+  private void setTraceIdAttributesPart(
+      AttributesBuilder builder,
+      String[] traceIdParts,
+      int indexPartsToCheck,
+      int indexTelemetryKeysToUse) {
+    if ((indexPartsToCheck < traceIdParts.length) && !traceIdParts[indexPartsToCheck].isEmpty()) {
+      builder.put(
+          telemetryKeysTraceId.get(indexTelemetryKeysToUse), traceIdParts[indexPartsToCheck]);
+    }
+  }
+
+  private void setTraceIdAttributes(AttributesBuilder builder) {
+    if ((this.traceId != null) && !this.traceId.isEmpty()) {
+      if (this.traceId.toLowerCase().indexOf("dataflow:") >= 0) {
+        String[] traceIdParts = this.traceId.split(":", 8);
+        for (int i = 0; i < traceIdParts.length; i++) {
+          if (traceIdParts[i].toLowerCase().indexOf("dataflow") >= 0) {
```

Comment:
Why we need a loop here? Can we remove L326 and just start with traceIdParts[0]

Reply:
There are cases where the "dataflow" value can appear in the first or the second part. Here are some examples of traceId values taken from actual logs:

Dataflow:2024-04-30_08_42_19-12633720189001946338
java-streamwriter:HEAD+20240508-1544 Dataflow:monorail-h-multi:2024-05-13_19_05_36-9076476533873711727:5306357987650008794
java-streamwriter Dataflow:pubsub-to-bq-staging-tongruil-1024-static:2024-05-14_15_13_14-5530509399715326669:7308531029326585036
java-streamwriter:HEAD+20240508-1544 Dataflow:monorail-c-multi:2024-05-08_11_44_34-6968230696879535523:2379627307761333914

Reply:
O, I think I know why, the HEAD+20240508-1544 is supposed to be the version number but obviously it is a wrong value (could also be the head you look at are from head client libs).... How about do a pattern match of "XXXX Dataflow:" and put everything in the first custom metrics and process the followings with ":".

Reply:
Done.